### DISCUSSION TITLE
How to use the logger when working on Stride?

### DISCUSSION THREAD

**Discussion by tebjan** [stride3d/stride#1032] [Q&A] [answered]
Asked in Discord:

How do I go about logging messages to the GameStudio output window from within source code files? Trying:

```
var logger = GlobalLogger.GetLogger("EditorModule");
logger.Warning("warning");
```

Didn't work. Nor did using `var logger = GlobalLogger.RegisteredLoggers[0]` instead.

Is there a way to log messages while in the GameStudio? I was looking into the rotation gizmo issues and the gizmo code would only run in GameStudio

**Comment by tebjan** [comment, marked as answer]
Answered by @manio143 :

When logging in your game a console window should pop up with the logs (not in GameStudio). Make sure you're running in Debug mode.
You're logging correctly (using GetLogger). If you want to do logs below Info level you need to set the `GlobalLogger.MinimumLevelEnabled`.
One more place where logs should appear is the Output/Debug window if you're using Visual Studio.

Are you modifying the source of Stride to log?
In that case you need to register a new debug log window, let me take a look
In [sources\editor\Stride.Editor\Build\GameStudioBuilderService.cs](https://github.com/stride3d/stride/blob/master/sources/editor/Stride.Editor/Build/GameStudioBuilderService.cs) you can register a debug log page
`effectCompilerServiceDebugPage = EditorDebugTools.CreateLogDebugPage(GlobalLogger.GetLogger("EditorModule"), "EditorModule");`
Then you access it with Help > Show Debug Window and there will be a new tab EditorModule.

**Comment by Redwarx008** [comment]
As mentioned previously, if you wish to print logs within the Studio, a convenient approach is to output them to the built-in Log Page.
``` c#
private static readonly Logger Log = GlobalLogger.GetLogger("Quantum");
Log.Debug();
```
